core/src/main/java/org/bitcoinj/core/PeerGroup.java

```diff
@@ -1128,7 +1128,7 @@ private boolean maybeCheckForLocalhostPeer() {
             // Do a fast blocking connect to see if anything is listening.
             try (Socket socket = new Socket()) {
                 socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), params.getPort()),
-                        vConnectTimeoutMillis);
+                        (int) vConnectTimeout.toMillis());
```

Comment:
Should you use `Math.toIntExact()` here to be safe?

Reply:
Good idea! Pushed a commit on top.